### ISSUE TITLE
Test application failure - PMIX ERROR: UNPACK-PAST-END in pmix_event_notification.c 

### ISSUE THREAD

**Issue by drwootton** [openpmix/openpmix#2017]
### What version of the PMIx Reference Library are you using? 
PMIx and PRRTE master as of 1/12/21 afternoon

### Describe how PMIx was installed 
Built from master branch source

### Please describe the system on which you are running

* Operating system/version:
RHEL 7.6
* Computer hardware:
Single Power 8 node, 22 cores, 176 threads
* Network type:
localhost
-----------------------------

## Details of the problem
I started a prte server with the **--system-server** flag then ran the PRRTE debugger direct example as **./direct**
This failed with two messages, one from each hello process spawned by direct
~~~
[f8n07:99586] PMIX ERROR: UNPACK-PAST-END in file event/pmix_event_notification.c at line 113
~~~
I modified the hello application to print its pid then sleep for 30 seconds, and modified direct to only spawn 1 application process.

I ran direct again, and attached with gdb to the sleeping hello process, set a breakpoint at pmix_event_notification.c:113 and resumed execution.

At the breakpoint, I got the traceback
~~~
#0  notify_event_cbfunc (pr=0x242f8940, hdr=0x2000010be2d0,
    buf=0x2000010be258, cbdata=0x24349be0)
    at event/pmix_event_notification.c:113
#1  0x00002000002d4168 in pmix_ptl_base_lost_connection (peer=0x2433a3c0,
    err=-61) at base/ptl_base_sendrecv.c:226
#2  0x00002000002d6134 in pmix_ptl_base_recv_handler (sd=7, flags=2,
    cbdata=0x2433a3c0) at base/ptl_base_sendrecv.c:607
#3  0x000020000008e038 in event_base_loop () from /lib64/libevent_core-2.0.so.5
#4  0x000020000021dacc in progress_engine (obj=0x242f54d8)
    at runtime/pmix_progress_threads.c:234
#5  0x00002000003e8b94 in start_thread () from /lib64/libpthread.so.0
#6  0x00002000005485f4 in clone () from /lib64/libc.so.6
~~~

The console output with direct modified to run a single task is
~~~
 f8n07:debugger> direct
Debugger ns prte-f8n0798705.13 rank 0 pid 101440: Running
Key pmix.qry.spawn Type PMIX_STRING(3)
Key pmix.qry.debug Type PMIX_STRING(3)
Debugger: spawning hello
Debugger: Registered for termination on nspace prte-f8n0798705.14
Key pmix.qry.ptable Type PMIX_DATA_ARRAY(39)
Received proc table for 1 procs
Debugger: spawning ./daemon
Debugger: Registered for termination on nspace prte-f8n0798705.15
Debugger daemon ns prte-f8n0798705.15 rank 0 pid 101443: Running
[prte-f8n0798705.15:0:101443] Debugging prte-f8n0798705.14
[prte-f8n0798705.15:0:101443] my local rank 0
[prte-f8n0798705.15:0:101443] registering for termination of prte-f8n0798705.14
Transferring pmix.qry.lptable
[prte-f8n0798705.15:0:101443] Local proctable received
[prte-f8n0798705.15:0:101443] Sending release
Client ns prte-f8n0798705.14 rank 0 pid 101442: Running on host f8n07 localrank 0
Client ns prte-f8n0798705.14 rank 0: Finalizing
[f8n07:101442] PMIX ERROR: UNPACK-PAST-END in file event/pmix_event_notification.c at line 113
Client ns prte-f8n0798705.14 rank 0:PMIx_Finalize successfully completed
DEBUGGER NOTIFIED THAT JOB prte-f8n0798705.14 TERMINATED
DEBUGGER DAEMON prte-f8n0798705.15 NOTIFIED THAT JOB TERMINATED - AFFECTED prte-f8n0798705.14
Debugger daemon ns prte-f8n0798705.15 rank 0 pid 101443: Finalizing
Debugger daemon ns prte-f8n0798705.15 rank 0 pid 101443:PMIx_Finalize successfully completed
DEBUGGER NOTIFIED THAT JOB prte-f8n0798705.15 TERMINATED
~~~
It appears the hello process terminated, and that processing to handle that got a bad message packet and failed.

This may be due to recent code changes. I do not get an error with PMIx source updated sometime in the last week.

**Comment by drwootton** [comment]
Naive question, looking at code around line 226 in ptl_base_sendrecv.c it looks like the code is setting up a minimal message to pass to notify_event_cbfunc. Could something be missing in that setup?

**Comment by rhc54** [comment]
This is a combination of things, I suspect. First, we shouldn't be trying to send a "lost connection" message to our server (whomever that is) because that is likely to be the connection that was lost. Second, we specifically flag that minimal message by assigning zero bytes to the body of it in the header, so we shouldn't be trying to unpack it at all.

I've fixed those two code paths to silence the warning. The next question, of course, is why it was being generated 😄

**Comment by rhc54** [comment]
@drwootton I'm unable to replicate an issue with current head of both PMIx and PRRTE master - the `direct` example runs just fine. Can you confirm?

**Comment by drwootton** [comment]
I built from current master. I don't see the error message now, with direct and direct co-spawn. This was a solid failure yesterday, where I ran direct maybe a dozen times and it issued the message every time.

However, since pull request #2018 is merged to master, the message should be gone anyway. Other than that, the messages displayed by direct looked reasonable, and it looked like the hello target had terminated and appropriate notifications sent/handled.

The way I found this was that I was trying to push changes for the CI test framework I have been working on, and the CI test failed, flagging a difference when comparing against a previously generated baseline file.
